src/GeneratedHydrator/CodeGenerator/Visitor/HydratorMethodsVisitor.php

```diff
@@ -50,6 +64,93 @@ public function __construct(ReflectionClass $reflectedClass)
     }
 
     /**
+     * Returns an array with properties as keys and hydrate/extract information
+     * as values.
+     *
+     * @param type $allowedProperties
+     */
+    private function expandAllowedProperties($options)
+    {
+        $allowedProperties = [];
+        $propertyNames = array_map(function($prop) {
+            return $prop->name;
+        }, $this->reflectedClass->getProperties());
+
+        if (! isset($options[static::OPTION_ALLOWED_PROPERTIES])) {
+            foreach ($propertyNames as $propertyName) {
+                $allowedProperties[$propertyName] = [
+                    'extract' => true,
+                    'hydrate' => true
+                ];
+            }
+
+            return $allowedProperties;
+        }
+
+        if (! is_array($options[static::OPTION_ALLOWED_PROPERTIES])) {
+            throw new \InvalidArgumentException(sprintf('OPTION_ALLOWED_PROPERTIES is given but it\'s value is of type %s which should be an array.', gettype($options[static::OPTION_ALLOWED_PROPERTIES])));
+        }
+
+        foreach ($options[static::OPTION_ALLOWED_PROPERTIES] as $k => $v) {
+            // simple format
+            if (is_int($k)) {
+                if (! is_string($v)) {
+                    throw new \InvalidArgumentException(sprintf('Invalid value of type %s found on index %s, expected a string.', gettype($v), $k));
```

Comment:
Use custom exceptions for this. Named exception constructors is what I'd suggest (see http://rosstuck.com/formatting-exception-messages/ )